src/components/WorkQueueCombobox.vue

```diff
@@ -49,20 +51,68 @@
   const api = useWorkspaceApi()
   const workQueuesSubscription = useSubscription(api.workQueues.getWorkQueues, [{}])
   const workQueues = computed(() => workQueuesSubscription.response ?? [])
-  const options = computed<SelectOption[]>(() => {
-    const options: SelectOption[] = workQueues.value.map(workQueue => ({
-      // Any consumers of the work queue should subscribe to it by name and not id
-      value: workQueue.name,
-      label: workQueue.name,
-    }))
+  const options = computed<SelectOptionGroup[]>(() => {
+    const workQueuesGroupedByWorkPool = workQueues.value.reduce<Map<string, WorkQueue[]>>((acc, workQueue) => {
+      // Filter options by work pool id or name
+      if (props.workPoolFilter?.id || props.workPoolFilter?.name) {
+        if (!props.workPoolFilter.id?.includes(workQueue.workPoolId) && !(workQueue.workPoolName && props.workPoolFilter.name?.includes(workQueue.workPoolName))) {
+          return acc
+        }
+      }
+
+      const workPoolName = workQueue.workPoolName ?? 'No work pool'
+      acc.set(workPoolName, (acc.get(workPoolName) ?? []).concat(workQueue))
+      return acc
+    }, new Map())
+
+    const options: SelectOptionGroup[] = []
+    for (const [workPoolName, workQueues] of workQueuesGroupedByWorkPool.entries()) {
+      options.push({
+        label: workPoolName,
+        options: workQueues.map(workQueue => ({
+          value: workQueue.id,
+          label: workQueue.name,
+        })),
+      })
+    }
 
     if (props.allowUnset) {
       options.unshift({
-        value: null,
-        label: 'None',
+        options: [{ value: null, label: 'None' }],
+        label: '',
       })
     }
 
     return options
   })
+
+  // Remove previously selected work queues that are no longer options
+  // when the options change
+  watch(options, (newValue, oldValue) => {
```

Comment:
Can you expound on why this is necessary? We don't have anything like this in our other combobox components.

Reply:
<img width="805" alt="image" src="https://github.com/PrefectHQ/prefect-ui-library/assets/22418768/921dc2e9-4ee7-446a-b8db-521db61ca405">

In that example ^, 
1) I pick some work queues that are across work pools
2) I decide no only queues from the "collins little minions" work pool
3)
    a) Without this, the combobox filters down the selectable options - one of which was previously selected and now it shows up as an empty tag without any content
    <img width="801" alt="image" src="https://github.com/PrefectHQ/prefect-ui-library/assets/22418768/c041f198-3489-4363-829e-b612f5cf9d08">
b) With this, the behavior is (imo) more consistent. if the value contains an option that's not a valid option after the options have changed, it's removed.

Reply:
hmm right because this combobox can filter by work pool which is kinda special compared to our other comboboxes. I'd probably just add an `@update:modelValue` to the work pool combobox and clear out the work queues when it changes. But I see the use case. Thanks for the explanation.

Reply:
yeah, I see your point 👍